### PR TITLE
Removing Unused Arguments(Minor Issues)

### PR DESCRIPTION
**Pull request by dhruvgoel2295**
These unused arguments to methods were listed under Minor Issues.

### CODE REVIEW

File: sunpy/visualization/mapcubeanimator.py [hunk 1]
```diff
@@ -65,7 +65,7 @@ def __init__(self, mapcube, annotate=True, **kwargs):
         if annotate:
             self._annotate_plot(0)
 
-    def updatefig(self, val, im, slider):
+    def updatefig(self, val, im, *args,**kwargs):
```
Comment (Cadair) [inline sunpy/visualization/mapcubeanimator.py]: I am pretty sure this will break things, the slider arg is part of the API.

Comment (Cadair) [inline sunpy/visualization/mapcubeanimator.py]: by that I mean this method can be overriden in superclasses which can do something with the slider argument.

**Comment by Cadair** [comment]
thanks for the contibution!

apart from the mapcubeanimator.py one, which I think is used outside of SunPy, the others are probably good. In the cases where only keyword arguments existed before, there is no need to add the `*args` and the same goes the other way, i.e. if there were no keyword arguments before then `*kwargs` does not need to be added.

**Comment by dhruvgoel2295** [comment]
Thanks for the feedback. Will keep that in mind for future contributions.

File: sunpy/lightcurve/sources/rhessi.py [hunk 2]
```diff
@@ -51,7 +51,7 @@ class RHESSISummaryLightCurve(LightCurve):
     * Mission Paper `<http://link.springer.com/article/10.1023%2FA%3A1022428818870>`_
     """
 
-    def peek(self, title="RHESSI Observing Summary Count Rate", **kwargs):
+    def peek(self, *args, **kwargs):
```
Comment (Cadair) [inline sunpy/lightcurve/sources/rhessi.py]: I don't think this needs `*args` either.

File: sunpy/data/_sample.py [hunk 3]
```diff
@@ -56,7 +56,7 @@
     sample_files[key] = os.path.abspath(os.path.join(sampledata_dir, _files[key][0]))
 
 
-def download_sample_data(progress=True, overwrite=True):
+def download_sample_data(overwrite=True,**kwargs):
```
Comment (Cadair) [inline sunpy/data/_sample.py]: progress needs removing from the docstring.

File: sunpy/lightcurve/sources/rhessi.py [hunk 4]
```diff
@@ -51,7 +51,7 @@ class RHESSISummaryLightCurve(LightCurve):
     * Mission Paper `<http://link.springer.com/article/10.1023%2FA%3A1022428818870>`_
     """
 
-    def peek(self, title="RHESSI Observing Summary Count Rate", **kwargs):
+    def peek(self , **kwargs):
```
Comment (Cadair) [inline sunpy/lightcurve/sources/rhessi.py]: no space before the comma.

**Comment by nabobalis** [comment]
Replaced with #2127

**Pull request closed without merging**